cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/messaging/aws/sqs/SqsTypedMessageQueue.java

```diff
@@ -63,4 +72,40 @@ protected TypedMessage toMessage(final com.amazonaws.services.sqs.model.Message
         }
     }
 
+    @Override
+    public List<TypedMessage> receive() throws MessageReceiveException {
+        final List<TypedMessage> receivedMessages = super.receive();
+        logReceivedMessages(receivedMessages);
+        return receivedMessages;
+    }
+
+    @Override
+    public List<TypedMessage> receive(final int waitTimeSeconds, final int maxMessages) throws MessageReceiveException {
```

Comment:
While this will correctly log all received messages, there's some more thinking needed to prevent the same message being logged twice. This is because inbound messages for domain events actually are forwarded from the event queue to *another* queue according to priority (either low or high priority). A possible solution is to add conditional logic here (based on an additional configuration parameter `logReceivedMessages` of this `MessageQueue`) which determines if the log statement is executed. Perhaps another solution is to configure the appenders somehow to only log certain queues.